### PR TITLE
Issue#442 url formfield

### PR DESCRIPTION
**Pull request by Hugo-Hontans**
Create url-form-field and implement ControlValueAccessor

### CODE REVIEW

File: libs/auth/src/lib/root/root.component.html [hunk 1]
```diff
@@ -3,8 +3,8 @@
   <section testId="signin" id="signin" fxLayout="column" fxLayoutAlign="start center">
     <h1>Sign in</h1>
     <form [formGroup]="signinForm" (ngSubmit)="signin()" fxLayout="column">
-      <input-email [form]="signinForm"></input-email>
-      <input-password [form]="signinForm" [showHint]=false></input-password>
+      <email-form-field [form]="signinForm"></email-form-field>
```
Comment (GrandSchtroumpf) [inline libs/auth/src/lib/root/root.component.html]: I thought you didn't need the `form` anymore ...

File: libs/movie/src/lib/movie/components/movie-form/movie-form-promotional/movie-form-promotional.component.html [hunk 2]
```diff
@@ -29,25 +29,28 @@ <h3> Images </h3>
...
-      </button>
-    </div>
+  <div formArrayName="promotionalElements" >
+    <ng-container *ngFor="let element of form.promotionalElements.controls; let i = index;">
+      <div [formGroupName]="i" class="full-width">
+        <mat-form-field class="small-width small-margin">
+          <input matInput formControlName="label" placeholder="Label">
+        </mat-form-field>
+        <url-form-field class="small-width small-margin" formControlName="url" placeholder="https://"></url-form-field>
+        <!-- <mat-form-field class="small-width small-margin">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/movie-form/movie-form-promotional/movie-form-promotional.component.html]: Remove if not needed anymore

File: libs/movie/src/lib/movie/components/movie-form/movie.form.ts [hunk 3]
```diff
@@ -178,8 +178,11 @@ export class MovieForm extends FormEntity<Movie, MovieControl> {
   }
 
   public addPromotionalElement(): void {
-    const defaultFormGroup = { label: '', url: ''};
-    this.addFormControl(this.builder.group(defaultFormGroup), 'promotionalElements');
+    const control = new FormGroup({
+      label: new StringControl(''),
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/movie-form/movie.form.ts]: Get ride of all `StringControl` in a next PR.

File: libs/ui/src/lib/form/form-field/url-form-field/url-form-field.component.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  set value(value: string) {
+    if (value) {
+      this._value = value;
+      this.propagateChange(this._value);
+    }
+  }
+
+  public addEvent(event) {
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/form/form-field/url-form-field/url-form-field.component.ts]: Type event

File: libs/ui/src/lib/form/form-field/url-form-field/url-form-field.component.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    this.value = event.target.value;
+    this.propagateChange(this.value);
+  }
+
+  public writeValue(value: string) {
+    if (value !== undefined) {
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/form/form-field/url-form-field/url-form-field.component.ts]: Why not `if (value)` ?

File: libs/auth/src/lib/root/root.component.html (hunk 1 above)
Comment (Hugo-Hontans) [reply libs/auth/src/lib/root/root.component.html]: I wanted to validate url-form-field before implementing ControlValueAccessor on others

File: libs/movie/src/lib/movie/components/movie-form/movie.form.ts (hunk 3 above)
Comment (Hugo-Hontans) [reply libs/movie/src/lib/movie/components/movie-form/movie.form.ts]: ok, I will do it

File: libs/ui/src/lib/form/form-field/url-form-field/url-form-field.component.ts [hunk 6]
```diff
@@ -29,29 +32,31 @@ export class UrlFormFieldComponent implements ControlValueAccessor {
   }
 
   set value(value: string) {
-    if (value) {
+    if (value !== undefined) {
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/form/form-field/url-form-field/url-form-field.component.ts]: What if it's `null` or empty string ?

Comment (Hugo-Hontans) [reply libs/ui/src/lib/form/form-field/url-form-field/url-form-field.component.ts]: it's for set the value if it's an empty string yes, we need that

**Pull request merged**